### PR TITLE
KDD central server codebase

### PR DESCRIPTION
**Pull request by andreievg**
Sorry no link to an issue, this has been talked about in a few meeting and in dev process chat:

Resources:

https://github.com/msupply-foundation/open-msupply-internal/discussions/47
https://github.com/msupply-foundation/open-msupply-internal/discussions/48

### CODE REVIEW

**Review by jmbrunskill — APPROVED** [review]
Looks good to me @andreievg 
I've been developing with the option 1 approach and agree it's manageable from a developer perspective.
Adding another central server creates new complications of course but I don't think any other option solves this problem.

**Comment by andreievg** [comment]
Thanks for review @jmbrunskill, I am keen on another set of eyes, maybe @Chris-Petty can have a look, when he gets the chance

File: decisions/2024-03-18_central_server_code_base.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+_Pros:_
+
+- Easy to work with in dev (no re-compilation)
+
+_Cons:_
+
+- More configurations during install
+
+#### 1b - Differentiated by feature flag
```
Comment (Chris-Petty) [inline decisions/2024-03-18_central_server_code_base.md:56]: Could you clarify how this is different from Option 2? Is this a feature flag that compiles the same project with feature conditions throughout, while Option 2 is a different project importing shared libraries that is built completely separately? (I'm not sure what the right [rust jargon](https://google.github.io/comprehensive-rust/glossary.html) is here.)

**Review by Chris-Petty — APPROVED** [review]
Yea option 1 seems best for now, though I think option 2 while idealistically better, in my experience breaking things out at our scale doesn't seem to work suuuuper well. Option 3 нет!! Same reason as option 2

File: decisions/2024-03-18_central_server_code_base.md (hunk 1 above)
Comment (andreievg) [reply decisions/2024-03-18_central_server_code_base.md:56]: 54e359b18083c2a5898dee32e18a427f6a52eeef
 
Yep basically how you explained it. Option 2 -> composing two different apps from shared, logic is seperate. Option 1b -> Logic is branched in place, can build one or the other based on feature flag/config

**Comment by andreievg** [comment]
@jmbrunskill @Chris-Petty, thanks for your input, merging

**Comment by andreievg** [comment]
Actually not super urgent, will wait another week

**Pull request merged**